### DISCUSSION TITLE
"website"/"web site", "webpage"/"web page", and the general issue of wordlists

### DISCUSSION THREAD

**Discussion by wbamberg** [mdn/mdn-community#372] [Content] [answered]
This was raised in https://github.com/mdn/content/issues/26461.

We don't consistently use the same spelling for "website"/"web site" or "webpage"/"web page".

In https://github.com/mdn/content/pull/26566, @dawei-wang has a fix for this, which updates pages to use "website" and "webpage". This is technically correct, because [we point to Microsoft's style guide for places we don't express an opinion](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide#preferred_style_guides), and [the MS style guide wants "website" and "webpage"](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/w/web-world-wide-web-www).

But having asked a few people about this, most people (including me) prefer "website" and "web page", and that's also what [the Google style guide prefers](https://developers.google.com/style/word-list#web-page).

I don't think we should maintain our own wordlist for everything, so deferring to another list is a good plan. But we might want to express an opinion about some words.

So, options include:

1. Merge https://github.com/mdn/content/pull/26566, keeping our guidance to use the MS style guide here.
2. Add a section in https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide like "selective wordlist" or something, where we list words we care especially about, and then defer to the MS wordlist for the rest.
3. Optionally (2) and also switch to the Google wordlist?

**Comment by hamishwillee** [comment]
The [preferred dictionary](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide#preferred_dictionary) is below the style guide section. That is dictionary.com, and it prefers `website`, `web page` and so on. 
- Doesn't that trump the style guides for spelling preferences? (we should make that clear)

IMO better to avoid separate word lists if we can avoid it. 
If it were me I'd update the preferred dictionary section to have it first.
Then word something like:

> For questions about spelling, refer to [Dictionary.com](https://www.dictionary.com/). The spelling checker for this site uses American English. Where there is any ambiguity, use the terminology in the [Google style guide](https://developers.google.com/style/word-list#web-page).

(unless you want style guide first, then dictionary?)

**Reply by wbamberg** [reply, marked as answer]
OK so it turns out that I can't read our writing guidelines documentation. There is a "Spelling" H3: https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide#spelling which gives us [dictionary.com](https://www.dictionary.com/) as the standard, which in turn gives us "website" and "web page".

So... maybe there's nothing to do here?

**Reply by yarusome** [reply]
@wbamberg It suggests that this page is overly lengthy and monolithic to act as a useful guide, no?

**Reply by wbamberg** [reply]
Yeah I would agree with that, I saw your issue there (it would also help if H3 headings appeared in the ToC, but your point stands).

**Reply by hamishwillee** [reply]
Yes, nothing to do, except perhaps move the dictionary above the style guide and make it clear which has priority. 

Yes the page is large and monolithic. But I'm not sure splitting it would make things much more easy to find. H3 headings in TOC might be better ....

**Reply by dipikabh** [reply]
I had reorganized and restructured the Writing style guide in positive hopes that we'll land the H3 headings in page TOC. And as a stopgap measure, I added a TOC after each H2 section.

Instead of having an exhaustive word list, we can have a small selection of words (as an example) to highlight. I can add it to the [Terminology](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide#terminology) section if you'll agree. This can include other words that Hamish highlighted - web, WWW, internet.
/cc @wbamberg @hamishwillee

**Reply by wbamberg** [reply]
It's true, splitting the page brings its own set of problems.

> Instead of having an exhaustive word list, we can have a small selection of words (as an example) to highlight. I can add it to the [Terminology](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide#terminology) section if you'll agree. This can include other words that Hamish highlighted - web, WWW, internet.
/cc @wbamberg @hamishwillee

I don't feel strongly either way but am -1 on this. We already have an answer about how to spell these words (dictionary.com). Unless we want to diverge from dictionary.com for any words, we should just point to that. Otherwise:
- we add more length and complexity to an already long and complex page
- it is confusing, people will think "why have they listed _these_ words, when the spelling is the same as dictionary.com anyway"
- we will potentially get into situations where people keep adding more words here

The only thing I might do is remove https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide#preferred_dictionary, since it's already covered in https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide#spelling.

**Reply by hamishwillee** [reply]
I'm with @wbamberg ^^^^ less is more.  Proposed fixes for the "no brainer" bit here: https://github.com/mdn/content/pull/26657

Note, I changed "preferred" to "other" for the style guides.

@dipikabh I think you've done well with the infrastructure available. 
The reason I would prefer the monolith is that what preceded this was a crowd of orphan docs . Just as hard to find, but worse - everyone who wanted to add something could without thinking about whether it was actually needed or relevant. This is better from a docs management point of view, and at least gives us a chance of having a lean-and-mean guide that people might read.

**Comment by OnkarRuikar** [comment]
I've been changing `web site` to `website` in my PRs whenever I could. We are down to only 114 files now :)

> So... maybe there's nothing to do here?

- It is time we add the lint rule https://github.com/mdn/content/pull/26682
